t/node/route-status.t

```diff
@@ -0,0 +1,304 @@
+                    "host": "foo.com",
+                    "upstream": {
+                        "nodes": {
+                            "127.0.0.1:1980": 1
+                        },
+                        "type": "roundrobin"
+                    }
+                }]]
+                )
+
+            if code >= 300 then
+                ngx.status = code
+            end
+            ngx.say(body)
+        }
+    }
+--- yaml_config eval: $::yaml_config
+--- request
+GET /t
+--- response_body
+passed
+--- no_error_log
+[error]
+
+
+
+=== TEST 6: hit route
+--- request
+GET /hello
+--- yaml_config eval: $::yaml_config
+--- more_headers
+Host: foo.com
+--- response_body
+hello world
+--- no_error_log
+[error]
+
+
+
+=== TEST 7: disable route
+--- config
+    location /t {
+        content_by_lua_block {
+            local core = require("apisix.core")
+            local t = require("lib.test_admin").test
+
+            local data = {status = 0}
+
+            local code, body = t('/apisix/admin/routes/1',
+                ngx.HTTP_PATCH,
+                core.json.encode(data),
+                [[{
+                    "node": {
+                        "value": {
+                            "status": 0
+                        },
+                        "key": "/apisix/routes/1"
+                    },
+                    "action": "compareAndSwap"
+                }]]
+                )
+
+            if code >= 300 then
+                ngx.status = code
+            end
+            ngx.say(body)
+        }
+    }
+--- yaml_config eval: $::yaml_config
+--- request
+GET /t
+--- response_body
+passed
+--- no_error_log
+[error]
+
+
+
+=== TEST 8: route not found, failed by disable
+--- request
+GET /hello
+--- yaml_config eval: $::yaml_config
+--- more_headers
+Host: foo.com
+--- error_code: 404
+--- response_body
+{"error_msg":"404 Route Not Found"}
+--- no_error_log
+[error]
+
+
+
+=== TEST 9: specify an invalid status value
+--- config
+    location /t {
+        content_by_lua_block {
+            local t = require("lib.test_admin").test
+            local code, body = t('/apisix/admin/routes/1',
+                 ngx.HTTP_PUT,
+                 [[{
+                        "uri": "/hello",
+                        "status": 100,
+                        "upstream": {
+                            "nodes": {
+                                "127.0.0.1:1980": 1
+                            },
+                            "type": "roundrobin"
+                        }
+                }]]
+                )
+            if code >= 300 then
+                ngx.status = code
+            end
+            ngx.say(body)
+        }
+    }
+--- request
+GET /t
+--- error_code: 400
+--- response_body eval
+qr/\{"error_msg":"invalid configuration: property \\"status\\" validation failed: matches non of the enum values"\}/
+--- no_error_log
+[error]
+
+
+
+=== TEST 10: compatible with old rpote data in etcd which not has status
```

Comment:
`rpote` Is this a typo, it should be `route`, right?

Reply:
> `rpote` Is this a typo, it should be `route`, right?

yep